tests/unittests/sources/test_vmware.py

```diff
@@ -1639,22 +1678,20 @@ def test_marker_file_setup(self):
         self.assertTrue(os.path.exists(markerfilepath))
 
 
-def assert_metadata(test_obj, ds, metadata):
-    test_obj.assertEqual(metadata.get("instance-id"), ds.get_instance_id())
-    test_obj.assertEqual(
-        metadata.get("local-hostname"), ds.get_hostname().hostname
-    )
+def assert_metadata(ds, metadata):
+    assert metadata.get("instance-id") == ds.get_instance_id()
+    assert metadata.get("local-hostname") == ds.get_hostname().hostname
 
     expected_public_keys = metadata.get("public_keys")
     if not isinstance(expected_public_keys, list):
         expected_public_keys = [expected_public_keys]
 
-    test_obj.assertEqual(expected_public_keys, ds.get_public_ssh_keys())
-    test_obj.assertIsInstance(ds.get_public_ssh_keys(), list)
+    assert expected_public_keys == ds.get_public_ssh_keys()
+    assert isinstance(ds.get_public_ssh_keys(), list)
 
 
-def get_ds(temp_dir):
```

Comment:
Now that we are using the `fake_filesystem` fixture, no need to pass `tmpdir` as `fake_filesystem` prepends `tmpdir` in all relevant calls (mainly by  the `os` module)